### ISSUE TITLE
Issue of training the building block encoder

### ISSUE THREAD

**Issue by FrancisFengbin** [microsoft/MOFDiff#20]
Hello, I got an error that when running the `python mofdiff/scripts/train.py --config-name=bb > ./testmilnew1.out` . And the output is 
```
pytorch_lightning.utilities.exceptions.MisconfigurationException: ReduceLROnPlateau conditioned on metric val_loss which is not available. Available metrics are: ['train_num_atom_loss', 'train_num_atom_loss_step', 'train_num_cp_loss', 'train_num_cp_loss_step', 'train_diameter_loss', 'train_diameter_loss_step', 'train_id_loss', 'train_id_loss_step', 'train_loss', 'train_loss_step', 'z_norm', 'z_norm_step', 'train_num_atom_loss_epoch', 'train_num_cp_loss_epoch', 'train_diameter_loss_epoch', 'train_id_loss_epoch', 'train_loss_epoch', 'z_norm_epoch']. Condition can be set using `monitor` key in lr scheduler dict

```
It seems that ReduceLROnPlateau relies on a loss metric from the validation set, such as val_loss. But I found `# building block embedding space learning does not involve validation or testing.` in bb_encoder.py. So what can I do to deal with this bug? Thanks!

**Comment by nayoung10** [comment]
Hi, I encountered the same problem. Under bb_encoder.py, try fixing the return value of ```def configure_optimizers``` as follows:

`        return {
            "optimizer": opt, 
            "lr_scheduler": { 
                "scheduler": scheduler, 
                "strict": False,
                "monitor": "val_loss"
            },
        }
`

**Comment by FrancisFengbin** [comment]
@lkny123 Thanks! I'll give it a try.

**Comment by 13734215431** [comment]
Hello, after I trained the model using my own dataset, I encountered the problem of not finding bb_emb_space.pt when generating mof. There is no code to save the file in the project
